thirdeye-ui/src/app/utils/rest/rest.util.ts

```diff
@@ -13,21 +13,195 @@
-import { get, isEmpty } from "lodash";
+import i18next from "i18next";
+import { get } from "lodash";
+import { ErrorMessage } from "../../platform/components/notification-provider-v1/notification-provider-v1/notification-provider-v1.interfaces";
 
-export const getErrorMessages = (error: AxiosError): string[] => {
-    const errMsgs: string[] = [];
+export enum ErrorCode {
+    ERR_UNAUTHENTICATED = "ERR_UNAUTHENTICATED",
+    ERR_ALERT_PIPELINE_EXECUTION = "ERR_ALERT_PIPELINE_EXECUTION",
+    ERR_AUTH_SERVER_NOT_RESPONDING = "ERR_AUTH_SERVER_NOT_RESPONDING",
+    ERR_CONFIG = "ERR_CONFIG",
+    ERR_CRON_INVALID = "ERR_CRON_INVALID",
+    ERR_CRON_FREQUENCY_TOO_HIGH = "ERR_CRON_FREQUENCY_TOO_HIGH",
+    ERR_DATASET_NOT_FOUND = "ERR_DATASET_NOT_FOUND",
+    ERR_DATASET_NOT_FOUND_IN_NAMESPACE = "ERR_DATASET_NOT_FOUND_IN_NAMESPACE",
+    ERR_DATASOURCE_NOT_FOUND = "ERR_DATASOURCE_NOT_FOUND",
+    ERR_DATASOURCE_NOT_FOUND_IN_NAMESPACE = "ERR_DATASOURCE_NOT_FOUND_IN_NAMESPACE",
+    ERR_DATASOURCE_NOT_LOADED = "ERR_DATASOURCE_NOT_LOADED",
+    ERR_DATASOURCE_VALIDATION_FAILED = "ERR_DATASOURCE_VALIDATION_FAILED",
+    ERR_DATA_UNAVAILABLE = "ERR_DATA_UNAVAILABLE",
+    ERR_DETECTION_INTERVAL_COMPUTATION = "ERR_DETECTION_INTERVAL_COMPUTATION",
+    ERR_DUPLICATE_NAME = "ERR_DUPLICATE_NAME",
+    ERR_DUPLICATE_ENTITY = "ERR_DUPLICATE_ENTITY",
+    ERR_ID_UNEXPECTED_AT_CREATION = "ERR_ID_UNEXPECTED_AT_CREATION",
+    ERR_INVALID_JSON_FORMAT = "ERR_INVALID_JSON_FORMAT",
+    ERR_INVALID_QUERY_PARAM_OPERATOR = "ERR_INVALID_QUERY_PARAM_OPERATOR",
+    ERR_MISSING_CONFIGURATION_FIELD = "ERR_MISSING_CONFIGURATION_FIELD",
+    ERR_MISSING_ID = "ERR_MISSING_ID",
+    ERR_MISSING_NAME = "ERR_MISSING_NAME",
+    ERR_MULTIPLE_DATASETS_FOUND = "ERR_MULTIPLE_DATASETS_FOUND",
+    ERR_MULTIPLE_DATASOURCES_FOUND = "ERR_MULTIPLE_DATASOURCES_FOUND",
+    ERR_NOTIFICATION_DISPATCH = "ERR_NOTIFICATION_DISPATCH",
+    ERR_NOT_ENOUGH_DATA_FOR_RCA = "ERR_NOT_ENOUGH_DATA_FOR_RCA",
+    ERR_OBJECT_DOES_NOT_EXIST = "ERR_OBJECT_DOES_NOT_EXIST",
+    ERR_OBJECT_UNEXPECTED = "ERR_OBJECT_UNEXPECTED",
+    ERR_OPERATION_UNSUPPORTED = "ERR_OPERATION_UNSUPPORTED",
+    ERR_TEMPLATE_MISSING_PROPERTY = "ERR_TEMPLATE_MISSING_PROPERTY",
+    ERR_TIMEOUT = "ERR_TIMEOUT",
+    ERR_UNEXPECTED_QUERY_PARAM = "ERR_UNEXPECTED_QUERY_PARAM",
+    ERR_UNKNOWN = "ERR_UNKNOWN",
+    ERR_UNKNOWN_RCA_ALGORITHM = "ERR_UNKNOWN_RCA_ALGORITHM",
+    ERR_CALCITE_FILTERING = "ERR_CALCITE_FILTERING",
+    ERR_INVALID_PARAMS_COMPONENTS = "ERR_INVALID_PARAMS_COMPONENTS",
+    ERR_INVALID_DETECTION_REGRESSORS = "ERR_INVALID_DETECTION_REGRESSORS",
+    ERR_INVALID_SQL = "ERR_INVALID_SQL",
+    ERR_NEGATIVE_LIMIT_VALUE = "ERR_NEGATIVE_LIMIT_VALUE",
+    ERR_NEGATIVE_OFFSET_VALUE = "ERR_NEGATIVE_OFFSET_VALUE",
+    ERR_OFFSET_WITHOUT_LIMIT = "ERR_OFFSET_WITHOUT_LIMIT",
+}
+
+export const getErrorMessages = (error: AxiosError): ErrorMessage[] => {
+    const errorCodeMessageMap: Record<ErrorCode, string> = {
```

Comment:
Nitpick: 
Objects without any dependence on the function params or state can be declared outside the function for a slightly less memory overload